### PR TITLE
Added ability to start snapd with global config

### PR DESCRIPTION
**Pull request by tiffanyfj**

### CODE REVIEW

**Comment by geauxvirtual** [comment]
Global config should not be appended to what already exists in control. It should be a separate package that handles parsing a config file on start of snapd, and then config options for Scheduler, Rest API, and Control are passed their config options when configured and started by the daemon.

File: control/config.go [hunk 1]
```diff
@@ -27,12 +27,34 @@ import (
...
 
 	log "github.com/Sirupsen/logrus"
+	"github.com/codegangsta/cli"
 
 	"github.com/intelsdi-x/snap/core"
 	"github.com/intelsdi-x/snap/core/cdata"
 	"github.com/intelsdi-x/snap/core/ctypes"
 )
 
+type flagConfig struct {
```
Comment (jcooklin) [inline control/config.go]: Looks good.  If you move this struct and its three methods to to pkg (e.g. pkg/cli/flags.go) I think we can get it merged.

**Comment by geauxvirtual** [comment]
I don't see how this addresses global config by calling control.NewConfig. Global config is for all of snap, and should be a config package outside of control.

**Comment by tiffanyfj** [comment]
@geauxvirtual Thoughts now? Thanks.

**Comment by geauxvirtual** [comment]
Why was this merged without a LGTM?

**Comment by jcooklin** [comment]
Note:  It had a verbal LGTM during the planning meeting by myself and Dan (before @geauxvirtual joined). 

After debating the PR further #710 was opened to address the remaining suggestions.

**Pull request merged**